Controller/Adminhtml/Reindex/Save.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * Module Algolia Algoliasearch
+ */
+namespace Algolia\AlgoliaSearch\Controller\Adminhtml\Reindex;
+
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Backend\App\Action\Context;
+use Magento\Catalog\Model\ProductFactory;
+use Magento\Store\Model\StoreManagerInterface;
+use Algolia\AlgoliaSearch\Helper\Data as DataHelper;
+use Algolia\AlgoliaSearch\Helper\Entity\ProductHelper;
+use Algolia\AlgoliaSearch\Exception\AlgoliaReindexException;
+use Algolia\AlgoliaSearch\Exception\AlgoliaProductDisabledException;
+use Algolia\AlgoliaSearch\Exception\AlgoliaProductDeletedException;
+use Algolia\AlgoliaSearch\Exception\AlgoliaProductNotVisibleException;
+use Algolia\AlgoliaSearch\Exception\AlgoliaProductOutOfStockException;
+
+/**
+ * Class: Save
+ */
+class Save extends \Magento\Backend\App\Action
+{
+
+    const MAX_SKUS = 10;
+
+    /**
+     * @var ProductFactory
+     */
+    private $productFactory;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var DataHelper
+     */
+    private $dataHelper;
+
+    /**
+     * @var ProductHelper
+     */
+    private $productHelper;
+
+    /**
+     *
+     * @param Context               $context
+     * @param ProductFactory        $productFactory
+     * @param StoreManagerInterface $storeManager
+     * @param DataHelper            $dataHelper
+     * @param ProductHelper         $productHelper
+     */
+    public function __construct(
+        Context $context,
+        ProductFactory $productFactory,
+        StoreManagerInterface $storeManager,
+        DataHelper $dataHelper,
+        ProductHelper $productHelper
+    ) {
+        parent::__construct($context);
+        $this->productFactory = $productFactory;
+        $this->storeManager = $storeManager;
+        $this->dataHelper = $dataHelper;
+        $this->productHelper = $productHelper;
+    }
+
+    /**
+     * Execute the action
+     *
+     * @return \Magento\Framework\View\Result\Page
+     *
+     * @throws AlgoliaReindexException
+     */
+    public function execute()
+    {
+        /** @var \Magento\Backend\Model\View\Result\Redirect $resultRedirect */
+        $resultRedirect = $this->resultFactory->create(ResultFactory::TYPE_REDIRECT);
+        $resultRedirect->setPath('*/*/index');
+        $skus = preg_split("/(,|\n)/", $this->getRequest()->getParam('skus'));
+
+        $stores = $this->storeManager->getStores();
+
+        foreach ($stores as $storeId => $storeData) {
+            if ($this->dataHelper->isIndexingEnabled($storeId) === false) {
+                unset($stores[$storeId]);
+            }
+        }
+
+        if (empty($skus)) {
+            $this->messageManager->addErrorMessage(__('Please enter one or more sku(s)'));
+        }
+
+        if (count($skus) > self::MAX_SKUS) {
+            $this->messageManager->addErrorMessage(__('Please enter less than %1 sku(s)', self::MAX_SKUS));
```

Comment:
`sku` => `SKU`
How about message `The maximal number of SKU(s) is 10. Could you please remove some SKU(s) to fit into the limit?`